service/backend/pchain/construction.go

```diff
@@ -0,0 +1,333 @@
+	suggestedFee := mapper.AtomicAvaxAmount(feeAmount)
+	return suggestedFee, nil
+}
+
+func (b *Backend) ConstructionPayloads(ctx context.Context, req *types.ConstructionPayloadsRequest) (*types.ConstructionPayloadsResponse, *types.Error) {
+	builder := pTxBuilder{
+		avaxAssetID:  b.avaxAssetID,
+		codec:        b.codec,
+		codecVersion: b.codecVersion,
+	}
+	return common.BuildPayloads(builder, req)
+}
+
+func (b *Backend) ConstructionParse(ctx context.Context, req *types.ConstructionParseRequest) (*types.ConstructionParseResponse, *types.Error) {
+	rosettaTx, err := b.parsePayloadTxFromString(req.Transaction)
+	if err != nil {
+		return nil, service.WrapError(service.ErrInvalidInput, err)
+	}
+
+	hrp, err := mapper.GetHRP(req.NetworkIdentifier)
+	if err != nil {
+		return nil, service.WrapError(service.ErrInvalidInput, "incorrect network identifier")
+	}
+
+	chainIDs := map[string]string{}
+	if rosettaTx.DestinationChainID != nil {
+		chainIDs[rosettaTx.DestinationChainID.String()] = rosettaTx.DestinationChain
+	}
+
+	txParser := pTxParser{
+		hrp:      hrp,
+		chainIDs: chainIDs,
+	}
+
+	return common.Parse(txParser, rosettaTx, req.Signed)
+}
+
+func (b *Backend) ConstructionCombine(ctx context.Context, req *types.ConstructionCombineRequest) (*types.ConstructionCombineResponse, *types.Error) {
+	rosettaTx, err := b.parsePayloadTxFromString(req.UnsignedTransaction)
+	if err != nil {
+		return nil, service.WrapError(service.ErrInvalidInput, err)
+	}
+
+	return common.Combine(b, rosettaTx, req.Signatures)
+}
+
+func (b *Backend) CombineTx(tx common.AvaxTx, signatures []*types.Signature) (common.AvaxTx, *types.Error) {
+	pTx, ok := tx.(*pTx)
+	if !ok {
+		return nil, service.WrapError(service.ErrInvalidInput, "invalid transaction")
+	}
+
+	ins, err := getTxInputs(pTx.Tx.Unsigned)
+	if err != nil {
+		return nil, service.WrapError(service.ErrInvalidInput, err)
+	}
+
+	creds, err := common.BuildCredentialList(ins, signatures)
+	if err != nil {
+		return nil, service.WrapError(service.ErrInvalidInput, err)
+	}
+
+	unsignedBytes, err := pTx.Marshal()
+	if err != nil {
+		return nil, service.WrapError(service.ErrInvalidInput, err)
+	}
+
+	pTx.Tx.Creds = creds
+
+	signedBytes, err := pTx.Marshal()
+	if err != nil {
+		return nil, service.WrapError(service.ErrInvalidInput, err)
+	}
+
+	pTx.Tx.Initialize(unsignedBytes, signedBytes)
+
+	return pTx, nil
+}
+
+// getTxInputs fetches tx inputs based on the tx type.
+func getTxInputs(
+	unsignedTx txs.UnsignedTx,
+) ([]*avax.TransferableInput, error) {
+	switch utx := unsignedTx.(type) {
+	case *txs.AddValidatorTx:
+		return utx.Ins, nil
+	case *txs.AddSubnetValidatorTx:
+		return utx.Ins, nil
+	case *txs.AddDelegatorTx:
+		return utx.Ins, nil
+	case *txs.CreateChainTx:
+		return utx.Ins, nil
+	case *txs.CreateSubnetTx:
+		return utx.Ins, nil
+	case *txs.ImportTx:
+		return utx.ImportedInputs, nil
+	case *txs.ExportTx:
+		return utx.Ins, nil
+	default:
+		return nil, errUnknownTxType
```

Comment:
nit: I'd list explictly all of the transactions here and maybe find another error than `errUnknownTxType` for them